fg/main.go

```diff
@@ -0,0 +1,202 @@
+func (f {{ $flag.Name }}Flag) GetName() string {
+    return f.Name
+}
+
+// IsRequired returns whether or not the flag is required
+func (f {{ $flag.Name }}Flag) IsRequired() bool {
+    return f.Required
+}
+
+// {{ $flag.Name }} looks up the value of a local {{ $flag.Name }}Flag, returns
+// {{ $flag.ContextDefault }} if not found
+func (c *Context) {{ $flag.Name }}(name string) {{ if ne .ContextType "" }} {{ $flag.ContextType }} {{ else }} {{ $flag.Type }} {{- end }} {
+    return lookup{{ $flag.Name }}(name, c.flagSet)
+}
+
+// Global{{ $flag.Name }} looks up the value of a global {{ $flag.Name }}Flag, returns
+// {{ $flag.ContextDefault }} if not found
+func (c *Context) Global{{ $flag.Name }}(name string) {{ if ne .ContextType "" }} {{ $flag.ContextType }} {{ else }} {{ $flag.Type }} {{- end }} {
+    if fs := lookupGlobalFlagSet(name, c); fs != nil {
+        return lookup{{ $flag.Name }}(name, fs)
+    }
+    return {{ $flag.ContextDefault }}
+}
+
+func lookup{{ $flag.Name }}(name string, set *flag.FlagSet) {{ if ne .ContextType "" }} {{ $flag.ContextType }} {{ else }} {{ $flag.Type }} {{- end }} {
+    f := set.Lookup(name)
+    if f != nil {
+        {{ if ne .Parser "" }}parsed, err := {{ $flag.Parser }}{{ else }}parsed, err := f.Value, error(nil){{ end }}
+        if err != nil {
+            return {{ $flag.ContextDefault }}
+        }
+        {{ if ne .ParserCast "" }}return {{ $flag.ParserCast }}{{ else }}return parsed{{ end }}
+    }
+    return {{ $flag.ContextDefault }}
+}
+{{- end }}
+{{ else }}
+import (
+	"flag"
+    "github.com/urfave/cli"
+)
+
+{{ range $i, $flag := .Flags }}
+// {{ $flag.Name }}Flag is the flag type that wraps cli.{{ $flag.Name }}Flag to allow
+// for other values to be specified
+type {{ $flag.Name }}Flag struct {
+    cli.{{ $flag.Name }}Flag
+    set *flag.FlagSet
+}
+
+// New{{ $flag.Name }}Flag creates a new {{ $flag.Name }}Flag
+func New{{ $flag.Name }}Flag(fl cli.{{ $flag.Name }}Flag) *{{ $flag.Name }}Flag {
+    return &{{ $flag.Name }}Flag{ {{ $flag.Name }}Flag: fl, set: nil }
+}
+
+// Apply saves the flagSet for later usage calls, then calls
+// the wrapped {{ $flag.Name }}Flag.Apply
+func (f *{{ $flag.Name }}Flag) Apply(set *flag.FlagSet) {
+	f.set = set
+	f.{{ $flag.Name }}Flag.Apply(set)
+}
+
+// ApplyWithError saves the flagSet for later usage calls, then calls
+// the wrapped {{ $flag.Name }}Flag.ApplyWithError
+func (f *{{ $flag.Name }}Flag) ApplyWithError(set *flag.FlagSet) error {
+    f.set = set
+    return f.{{ $flag.Name }}Flag.ApplyWithError(set)
+}
+{{- end }}
+{{- end }}
+`
+
+func main() {
+	var packageName, inputPath, outputPath string
+
+	app := cli.NewApp()
+
+	app.Name = "fg"
+	app.Usage = "Generate flag type code!"
+	app.Version = "v0.1.0"
+
+	app.Flags = []cli.Flag{
+		cli.StringFlag{
+			Name:        "package, p",
+			Value:       "cli",
+			Usage:       "`PACKAGE` for which the flag types will be generated",
+			Destination: &packageName,
```

Comment:
⭐⭐ _(seriously consider)_

We only have the two packages yeah? Can we make this a string slice, and have it default to both `cli` and `altsrc`? (or whatever the two names are)

Reply:
I used this approach as this started as a port of the existing python script. I think having defaults makes more sense as:

- The use of this tool is highly local to `urfave/cli`
- we would be able to remove the current two //go generate from the codebase to only a single one
- would clean up the eventual build process by requiring only a single command to generate the necessary bits

Reply:
👍👍👍👍

Reply:
Done. All switches to accept input are now gone. We generate code for cli and altsrc for default